subprojects/signing/src/integTest/groovy/org/gradle/plugins/signing/SigningPublicationsIntegrationSpec.groovy

```diff
@@ -845,4 +845,78 @@ class SigningPublicationsIntegrationSpec extends SigningIntegrationSpec {
         file("build", "libs", "sign-1.0.jar.asc").text
         file("build", "publications", "mavenJava", "pom-default.xml.asc").text
     }
+
+    @Issue("https://github.com/gradle/gradle/issues/26091")
+    def "PublishToMaven task should depend on Sign task automatically"() {
+        given:
+        buildFile.delete()
+        buildKotlinFile << """
+            import org.gradle.jvm.tasks.Jar
+
+            plugins {
+                kotlin("multiplatform") version "1.9.0"
+                id("maven-publish")
+                id("signing")
+            }
+
+            kotlin {
+                jvm()
+                js { browser() }
+            }
+
+            repositories {
+                mavenCentral()
+            }
+
+            group = "com.example"
+            version = "0.0.1"
+
+            val javadocJar by tasks.registering(Jar::class) {
+                archiveClassifier.set("javadoc")
+            }
+
+            publishing {
+                publications.withType<MavenPublication> {
+                    artifact(javadocJar)
+                }
+
+                signing {
+                    useInMemoryPgpKeys(project.property("secretKey").toString(), project.property("password").toString())
+                    sign(publications)
+                }
```

Comment:
This part should be outside of the publishing block.

Reply:
line 885 refers to the variable **publications** defined in publishing block, so moving sign block outside publishing block will break the compilation.

Reply:
Use `publishing.publications` 🙃